build.rs

```diff
@@ -2,17 +2,44 @@ use cmake::Config;
 use std::{env, path::PathBuf};
 
 fn main() {
-    let dst = Config::new("libllvm-pdb-wrapper").build();
-    println!("cargo:rustc-link-search=native={}", dst.display());
-    println!("cargo:rustc-link-lib=static=llvm-pdb-wrapper");
+    println!("Remember to set the LIBCLANG_PATH and/or LLVM_DIR if needed");
+    if std::env::var_os("CARGO_FEATURE_LLVM_13").is_some() {
+        println!("specialbuold");
+        let dst = Config::new("libllvm-pdb-wrapper")
+            .define("LLVM_NEWER", "ON")
+            .build();
+        println!("cargo:rustc-link-search=native={}", dst.display());
+        println!("cargo:rerun-if-changed={}", dst.display());
+    } else {
+        let dst = Config::new("libllvm-pdb-wrapper").build();
+        println!("cargo:rustc-link-search=native={}", dst.display());
+        println!("cargo:rerun-if-changed={}", dst.display());
+    }
 
+    println!("cargo:rustc-link-lib=static=llvm-pdb-wrapper");
     println!("cargo:rustc-link-lib=llvm-pdb-wrapper");
-    println!("cargo:rustc-link-lib=LLVM-10");
-    println!("cargo:rustc-link-lib=stdc++");
+    if std::env::var_os("CARGO_FEATURE_LLVM_13").is_some() {
+        println!("cargo:rustc-link-lib=LLVMDebugInfoCodeView");
+        println!("cargo:rustc-link-lib=LLVMDebugInfoPDB");
+        println!("cargo:rustc-link-lib=LLVMDebugInfoMSF");
+        println!("cargo:rustc-link-lib=LLVMMC");
+        println!("cargo:rustc-link-lib=LLVMCore");
+        println!("cargo:rustc-link-lib=LLVMBinaryFormat");
+        println!("cargo:rustc-link-lib=LLVMRemarks");
+        println!("cargo:rustc-link-lib=LLVMBitstreamReader");
+        println!("cargo:rustc-link-lib=LLVMMCParser");
+        println!("cargo:rustc-link-lib=LLVMObject");
+        println!("cargo:rustc-link-lib=LLVMTextAPI");
+        println!("cargo:rustc-link-lib=LLVMBitReader");
+        println!("cargo:rustc-link-lib=LLVMSupport");
+        println!("cargo:rustc-link-lib=zlib");
```

Comment:
It may be best to figure out how to call llvm-config or the like to extract these link dependencies

Reply:
[This](https://doc.rust-lang.org/cargo/reference/build-script-examples.html#building-a-native-library) should cover invoking commands from build.rs

Reply:
I think ill leave the linking as is for now I cant seem to choose a good way to extract the libs from llvm-config.